### PR TITLE
[pkg/flare] Improve `zipFile` signature

### PR DESCRIPTION
**Pull request by levipe01**
### What does this PR do?

- Change zipFile signature to not require basename duplication
- Add tests for changed functions

### Motivation

- By ensuring that the source and destination files have the same basename we will reduce the chance that the programmer calling this function will make errors

### Possible Drawbacks / Trade-offs

none

### Describe how to test/QA your changes

- On linux generate a flare with the agent and ensure that `registry.json` and `version-history.json` have content
- On linux generate a flare with the security-agent and ensure that `kallsyms`, `kprobe_events`, `mountinfo`, `available_events`, & `available_filter_functions` files have content (note that `kprobe_events` may be empty)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/flare/archive.go [hunk 1]
```diff
@@ -645,15 +647,15 @@ func zipFile(originalPath, zippedPath string) error {
 }
 
 func zipRegistryJSON(tempDir, hostname string) error {
-	originalPath := filepath.Join(config.Datadog.GetString("logs_config.run_path"), "registry.json")
-	zippedPath := filepath.Join(tempDir, hostname, "registry.json")
-	return zipFile(originalPath, zippedPath)
+	originalPath := filepath.Join(config.Datadog.GetString("logs_config.run_path"))
```
Comment (hush-hush) [inline pkg/flare/archive.go:650]: I don't think we need the `filepath.Join` anymore.

File: pkg/flare/archive.go [hunk 2]
```diff
@@ -645,15 +647,15 @@ func zipFile(originalPath, zippedPath string) error {
...
+	targetPath := filepath.Join(tempDir, hostname)
+	return zipFile(originalPath, targetPath, "registry.json")
 }
 
 func zipVersionHistory(tempDir, hostname string) error {
-	originalPath := filepath.Join(config.Datadog.GetString("run_path"), "version-history.json")
-	zippedPath := filepath.Join(tempDir, hostname, "version-history.json")
-	return zipFile(originalPath, zippedPath)
+	originalPath := filepath.Join(config.Datadog.GetString("run_path"))
```
Comment (hush-hush) [inline pkg/flare/archive.go:656]: same here.

File: pkg/flare/archive.go [hunk 3]
```diff
@@ -774,9 +777,9 @@ func zipHealth(tempDir, hostname string) error {
 }
 
 func zipInstallInfo(tempDir, hostname string) error {
-	originalPath := filepath.Join(config.FileUsedDir(), "install_info")
-	zippedPath := filepath.Join(tempDir, hostname, "install_info")
-	return zipFile(originalPath, zippedPath)
+	originalPath := filepath.Join(config.FileUsedDir())
```
Comment (hush-hush) [inline pkg/flare/archive.go:780]: same here.

**Review by hush-hush — CHANGES_REQUESTED** [review]

**Comment by levipe01** [comment]
@hush-hush fixed your comments, do you mind taking another look?

**Review by hush-hush — APPROVED** [review]

**Pull request merged**